AK/BufferedStream.h

```diff
@@ -115,6 +115,53 @@ class BufferedHelper {
         return m_buffer.read(buffer);
     }
 
+    ErrorOr<StringView> read_line_with_resize(ByteBuffer& buffer)
+    {
+        return StringView { TRY(read_until_with_resize(buffer, "\n"sv)) };
+    }
+
+    ErrorOr<Bytes> read_until_with_resize(ByteBuffer& buffer, StringView candidate)
+    {
+        return read_until_any_of_with_resize(buffer, Array { candidate });
+    }
+
+    template<size_t N>
+    ErrorOr<Bytes> read_until_any_of_with_resize(ByteBuffer& buffer, Array<StringView, N> candidates)
+    {
+        if (!stream().is_open())
+            return Error::from_errno(ENOTCONN);
+
+        auto candidate = TRY(find_and_populate_until_any_of(candidates));
+
+        size_t bytes_read_to_user_buffer = 0;
+        while (!candidate.has_value()) {
+            if (m_buffer.used_space() == 0 && stream().is_eof()) {
+                if (bytes_read_to_user_buffer == 0) {
+                    // There are no bytes left in m_buffer or in the stream,
+                    // and nothing was read to the user supplied buffer,
+                    // so return an empty buffer.
+                    return Bytes {};
+                } else {
+                    return buffer.span().trim(bytes_read_to_user_buffer);
+                }
```

Comment:
These two branches end up being the same for our purposes (or rather, the second is a strict superset of the first), so we can just unify them (and gain some space for explanations in the process).

```suggestion
                // If we read to the very end of the buffered and unbuffered data,
                // then treat the remainder as a full line (the last one), even if it
                // doesn't end in the delimiter.
                return buffer.span().trim(bytes_read_to_user_buffer);
```